src/main/java/net/minecraftforge/network/simple/SimpleChannel.java

```diff
@@ -223,11 +246,44 @@ public MessageBuilder<MSG> noResponse()
             return this;
         }
 
-        public MessageBuilder<MSG> consumer(BiConsumer<MSG, Supplier<NetworkEvent.Context>> consumer) {
```

Comment:
There's no need to binary break here. Mark consumer as @Deprecated and redirect to consumerMainThread in it. Otherwise this looks fine. I need to look at encode/decode offthread too.